spring-cloud-gcp-pubsub/src/test/java/org/springframework/cloud/gcp/pubsub/converters/SimpleMessageConverterTest.java

```diff
@@ -39,10 +38,16 @@
 	@Test
 	public void testFromMessage() {
 		SimpleMessageConverter converter = new SimpleMessageConverter();
-		Message<?> message = new GenericMessage<Object>("test payload");
+
+		Map<String, Object> headers = new HashMap<>();
+		headers.put("h1", "value");
+		headers.put("h2", 23);
+		Message<?> message = new GenericMessage<Object>("test payload", headers);
 		PubsubMessage pubsubMessage = (PubsubMessage) converter.fromMessage(message,
 				PubsubMessage.class);
 		assertEquals("test payload", pubsubMessage.getData().toStringUtf8());
+		assertEquals("value", pubsubMessage.getAttributesOrDefault("h1", "def"));
+		assertEquals("23", pubsubMessage.getAttributesOrDefault("h2", "def"));
```

Comment:
That is good point! And right, we don't send them in other protocols, unless end-user requests that via `HeaderMapper`.